io/src/lib.rs

```diff
@@ -425,7 +427,8 @@ mod tests {
         assert_eq!(&buf, "16".as_bytes())
     }
 
-    /// A reader that always returns a fixed number of bytes counting up how many times `read` was
+    /// A reader provides an unlimited number of bytes but always returns a fixed number of bytes
```

Comment:
I think the correct word here is "limited", not "fixed". If the buffer is shorter the number of returned bytes is shorter too.